File: src/main/java/io/qross/sql/ForSelectLoop.java
package io.qross.sql;

import io.qross.util.DataSource;

public class ForSelectLoop {

    public Statement statement;
    public String[] variables;
    public String selectSQL;

    public ForSelectLoop(Statement statement, String variables, String selectSQL) {
        this.statement = statement;
        this.variables = Statement.removeVariableModifier(variables).split(",");
        this.selectSQL = selectSQL;
    }

    public ForLoopVariables computeMap(DataSource ds) {

        ForLoopVariables variables = new ForLoopVariables();

        ds.executeDataTable(selectSQL).forEach(variables::addRow);

        return variables;
    }
}

File: src/main/scala/io/qross/psql/ForSelectLoop.scala
package io.qross.psql

import io.qross.jdbc.DataSource
import io.qross.util.TypeExt._

class ForSelectLoop(statement: Statement, var variable: String, selectSQL: String) {
    private val variables = variable.removeVariableModifier().split(",")

    def computeMap(ds: DataSource): ForLoopVariables = {
        val variables = new ForLoopVariables()

        ds.executeDataTable(selectSQL).foreach(variables.addRow)

        variables
    }
}
